cpp/modmesh/buffer/SimpleArray.cpp

```diff
@@ -573,6 +573,44 @@ SimpleArrayPlex::~SimpleArrayPlex()
     m_has_instance_ownership = false;
 }
 
+/**
+ * Find the two bins that correspond to the median values for frequency-based median calculation.
+ * This function is used for small data types (uint8_t, int8_t, bool) where frequency counting
+ * is more efficient than sorting.
+ */
+template <typename A, typename T>
+void detail::SimpleArrayMixinCalculators<A, T>::find_two_bins(const uint32_t * freq, size_t n, int & bin1, int & bin2)
+{
+    const size_t k1 = (n - 1) / 2;
+    const size_t k2 = n / 2;
+
+    uint32_t cumulative = 0;
+    bin1 = -1;
+    bin2 = -1;
+
+    for (int i = 0; i < 256; ++i)
+    {
+        cumulative += freq[i];
+        if (bin1 < 0 && cumulative > k1)
+        {
+            bin1 = i;
+        }
+        if (bin2 < 0 && cumulative > k2)
+        {
+            bin2 = i;
+        }
+        if (bin1 >= 0 && bin2 >= 0)
+        {
+            break;
+        }
+    }
+}
+
+// Explicit instantiations for all SimpleArray types that use find_two_bins
+template void detail::SimpleArrayMixinCalculators<SimpleArray<bool>, bool>::find_two_bins(const uint32_t * freq, size_t n, int & bin1, int & bin2);
```

Comment:
Instantiating templates in an implementation (cpp) file smells bad.  It could imply something goes wrong.  But let's do it now and see what will happen.